portals/publisher/src/main/webapp/source/src/app/components/Apis/Create/GraphQL/Steps/ProvideGraphQL.jsx

```diff
@@ -106,31 +113,161 @@ export default function ProvideGraphQL(props) {
+        );
+    } else if (isValid.url !== undefined) {
+        if (isInvalidURL) {
+            urlStateEndAdornment = (
+                <InputAdornment position='end'>
+                    <ErrorOutlineIcon fontSize='large' color='error' />
+                </InputAdornment>
+            );
+        } else {
+            urlStateEndAdornment = (
+                <InputAdornment position='end' id='url-validated'>
+                    <CheckIcon fontSize='large' color='primary' />
+                </InputAdornment>
+            );
+        }
+    }
+
+    const debouncedValidateURLOrEndpoint = useCallback(
+        debounce((newURL) => {
+            const handleResponse = (response) => {
+                const {
+                    body: { graphQLInfo },
+                } = response;
+                const isValidURL = response.body.isValid;
+                if (isValidURL) {
+                    inputsDispatcher({ action: 'graphQLInfo', value: graphQLInfo });
+                    setValidity({ isValidURL, file: null });
+                } else {
+                    setValidity({ isValidURL, file: { message: 'GraphQL content validation failed!' } });
+                }
+                onValidate(isValidURL);
+                setIsValidating(false);
+            };
+
+            const handleError = (error) => {
+                setValidity({ url: { message: error.message } });
+                onValidate(false);
+                setIsValidating(false);
+                console.error(error);
+            }
+            
+            const validationFunction = inputType === ProvideGraphQL.INPUT_TYPES.URL
+                ? () => API.validateGraphQLByUrl(newURL)
+                : () => API.validateGraphQLByEndpoint(newURL, { useIntrospection: true });
+
+            validationFunction()
+                .then(handleResponse)
+                .catch(handleError);
+
+            if(inputType === ProvideGraphQL.INPUT_TYPES.ENDPOINT){
+                inputsDispatcher({ action: 'endpoint', value: newURL });
+            }
+        }, 750),
+        [inputType],
+    );
+
+    function validateURL(value) {
+        const state = APIValidation.url.required().validate(value).error;
+        if (state === null) {
+            setIsValidating(true);
+            debouncedValidateURLOrEndpoint(apiInputs.inputValue);
+        } else {
+            setValidity({ ...isValid, url: state });
+            onValidate(false);
+        }
+    }
+
     useEffect(() => {
         if (inputValue) {
-            onDrop([inputValue]);
+            if (inputType === ProvideGraphQL.INPUT_TYPES.FILE) {
+                onDrop([inputValue]);
+            } else if (inputType === ProvideGraphQL.INPUT_TYPES.URL ||
+                inputType === ProvideGraphQL.INPUT_TYPES.ENDPOINT) {
+                validateURL(inputValue);
+            }
         }
-    }, [inputValue]);
+    }, [inputType, inputValue]);
+
+    useEffect(() => {
+        reset();
+    }, [inputType]);
```

Comment:
Why do we need this useEffect? Doesn't the above useEffect depend on the same `inputType` value as well? Can't we call `reset()` there?

Reply:
If used with the other useEffect, it would run the reset() when inputValue is changed